HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+    ```
+
+
+* #### `GrafanaDatasourceUrlGetHandler`
+
+    ###### synopsis
+    Returns the URL of the Grafana datasource that Container JFR
+    is configured with (determined by the environment variable
+    `GRAFANA_DATASOURCE_URL`).
+
+    ###### request
+    `GET /api/v1/grafana_datasource_url`
+
+    ###### response
+    `200` - The body is `{"grafanaDatasourceUrl":"$URL"}`
+
+    `500` - Container JFR is not configured with a Grafana datasource.
+    Or there was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v1/grafana_datasource_url
+    {"grafanaDatasourceUrl":"http://localhost:8080"}
+    ```
+
+
+* #### `HealthGetHandler`
+
+    ###### synopsis
+    Returns whether or not the Grafana datasource and Grafana dashbaord
+    that Container JFR is configured with are running properly.
+
+    ###### request
+    `GET /health`
+
+    ###### response
+    `200` - The body is
+    `{"datasourceAvailable":$DATASOURCE_AVAILABLE,"dashboardAvailable":$DASHBOARD_AVAILABLE}`.
+
+    `$DATASOURCE_AVAILABLE` is `true` if  Container JFR is configured with a
+    Grafana datasource and that datasource responds to a `GET` request
+    with a `200`, and it is `false` otherwise.
+
+    `$DASHBOARD_AVAILABLE` is `true` if  Container JFR is configured with a
+    Grafana dashboard and that dashboard responds to a `GET` request
+    with a `200`, and it is `false` otherwise.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/health
+    {"dashboardAvailable":false,"datasourceAvailable":false}
+    ```
+
+
+* #### `StaticAssetsGetHandler`
+
+    ###### synopsis
+    Returns a static asset of the Container JFR web client.
+
+    ###### request
+    `GET /*`
+
+    The path should be the directory path of the desired asset,
+    assuming a root of
+    `target/assets/app/resources/com/redhat/rhjmc/containerjfr/net/web/`.
+
+    ###### response
+
+    `200` - The body is the requested asset. Note that if the requested asset
+    does not exist or the path is invalid, the request will not be handled here
+    and instead  will be routed to
+    [`WebClientAssetsGetHandler`](#WebClientAssetsGetHandler).
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/images/pfbg_992@2x.jpg --output pfbg_992@2x.jpg
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  574k  100  574k    0     0  43.1M      0 --:--:-- --:--:-- --:--:-- 43.1M
+    ```
+
+
+* #### `TargetsGetHandler`
+
+    ###### synopsis
+    Scans for and returns discoverable target JVMs.
+
+    ###### request
+    `GET /api/v1/targets`
+
+    ###### response
+    `200` - The body consists of a comma-seperated list of targets,
+    enclosed in square brackets.
```

Comment:
This `comma-separated list of targets, enclosed in square brackets` can be more concisely described as `a JSON array of target objects`